addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/HandlerFactory.java

```diff
@@ -79,12 +89,32 @@ public void unsetHttpClientFactory(HttpClientFactory factory) {
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-
         if (BRIDGE_TYPE.equals(thingTypeUID)) {
-            return new DeconzBridgeHandler((Bridge) thing, webSocketFactory,
+            DeconzBridgeHandler handler = new DeconzBridgeHandler((Bridge) thing, webSocketFactory,
                     new AsyncHttpClient(httpClientFactory.getCommonHttpClient()));
+            ThingDiscoveryService discoveryService = new ThingDiscoveryService(handler);
```

Comment:
Is `getServices()` not working? I mean it was meant to be used for such use cases

Reply:
The problem why it was no working were theses lines of code:

https://github.com/openhab/openhab2-addons/blob/2b9c68e54dc037073beb0ec0d4f1dd5214a3aa60/addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java#L163-L166

The `thingDiscoveryService` has never been assigned to the handler. The implemented solution was the only way I could imagine how to set the missing reference. Is there a better way to do it?

Reply:
It's a bit of a two way reference situation, I agree. But the pattern with `getServices()` requires this.

What happens is:
* the framework requests the service list via `getServices()`. 
* It starts all those services for the ThingHandler and calls `setThingHandler(handler)`
* And in that method we set the discoveryService reference within the handler. See:

https://github.com/openhab/openhab2-addons/blob/826b0f41e7d3920e712c7c63c6e91d2fe11875d4/addons/binding/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/discovery/ThingDiscoveryService.java#L106